unstract/connectors/src/unstract/connectors/filesystems/box/box.py

```
@@ -131,7 +132,48 @@ def is_dir_by_metadata(self, metadata: dict[str, Any]) -> bool:
+            return None
+        try:
+            if isinstance(value, datetime):
+                return value if value.tzinfo else value.replace(tzinfo=UTC)
+            if isinstance(value, (int, float)):
+                return datetime.fromtimestamp(value, tz=UTC)
+            if isinstance(value, str):
+                # Normalize 'Z' suffix to '+00:00' for fromisoformat compatibility
+                v = value.strip().replace("Z", "+00:00")
+                dt = datetime.fromisoformat(v)
+                return dt if dt.tzinfo else dt.replace(tzinfo=UTC)
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)